File: app/src/main/java/com/mcoptimizer/launcher/LauncherActivity.java
// MinecraftMobileOptimizer/app/src/main/java/com/mcoptimizer/launcher/LauncherActivity.java
package com.mcoptimizer.launcher;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import com.mcoptimizer.R;
import com.mcoptimizer.jni.NativeBridge;

public class LauncherActivity extends AppCompatActivity {
    
    private TextView statusText;
    private final Handler handler = new Handler(Looper.getMainLooper());
    private volatile boolean running = true;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_launcher);
        
        statusText = findViewById(R.id.statusText);
        
        int targetFps = getIntent().getIntExtra("TARGET_FPS", 60);
        float renderScale = getIntent().getFloatExtra("RENDER_SCALE", 1.0f);
        
        statusText.setText("MC 1.21.11 | Target: " + targetFps + " FPS | Scale: " + 
                          String.format("%.2f", renderScale));
        
        startGameLoop();
    }
    
    private void startGameLoop() {
        new Thread(() -> {
            while (running && !isDestroyed()) {
                NativeBridge.beginFrame();
                try { Thread.sleep(8); } catch (InterruptedException e) { break; }
                NativeBridge.endFrame();
            }
        }).start();
    }
    
    @Override protected void onDestroy() {
        running = false;
        super.onDestroy();
        NativeBridge.shutdown();
    }
}

File: app/src/main/java/com/mcoptimizer/launcher/LauncherActivity.kt
package com.mcoptimizer.launcher

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView
import android.widget.Button
import android.util.Log
import com.mcoptimizer.R
import com.mcoptimizer.optimizer.RenderOptimizer
import com.mcoptimizer.optimizer.MemoryManager
import com.mcoptimizer.optimizer.PerformanceMonitor

class LauncherActivity : AppCompatActivity() {
    companion object {
        private const val TAG = "LauncherActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_launcher)

        val statusText = findViewById<TextView>(R.id.txt_status)
        val installButton = findViewById<Button>(R.id.btn_install_game)
        val launchButton = findViewById<Button>(R.id.btn_launch)

        // Initialize performance monitor
        PerformanceMonitor.startMonitoring()

        installButton.setOnClickListener {
            GameInstaller.installGame(this) { success ->
                runOnUiThread {
                    if (success) {
                        statusText.text = "Game installed successfully!"
                        launchButton.isEnabled = true
                    } else {
                        statusText.text = "Installation failed"
                    }
                }
            }
        }

        launchButton.setOnClickListener {
            Log.d(TAG, "Launching game with optimizations")
            RenderOptimizer.applyOptimizations()
            MemoryManager.clearCache()
            GameInstaller.launchGame()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        PerformanceMonitor.stopMonitoring()
    }
}
